dialer.go

```diff
@@ -143,13 +151,9 @@ func NewDialer(ctx context.Context, opts ...DialerOption) (*Dialer, error) {
 		opt(&dialCfg)
 	}
 
-	if err := trace.InitMetrics(); err != nil {
-		// This error means the internal metric configuration is incorrect and
-		// should never be surfaced to callers, as there's nothing actionable
-		// for a caller to do. Ignoring the error seems worse and so we return
-		// it.
-		return nil, err
-	}
+	// The error returned indicates a configuration error inside the library and
+	// will be caught by tests. So we safely ignore it here.
+	mc, _ := trace.NewMetricsCollector()
```

Comment:
I'm not sure it's safe to ignore this error. Assuming the tests catch everything also assumes that the version of the dependency is the same as what we've tested. I don't think that's the case here?

I guess it seems like: if we ignore the error and it happens, it blows up with probably a nil reference? If we don't ignore the error and it happens, they at least get an error with what's wrong?

Reply:
Yeah, it's cheap to just return it, even if it's unlikely or even impossible.